state-chain/pallets/cf-validator/src/lib.rs

```diff
@@ -801,6 +862,206 @@ pub mod pallet {
+
+			ensure!(account_id == operator || account_id == validator, Error::<T>::NotAuthorized);
+
+			ManagedValidators::<T>::remove(validator);
+
+			Ok(())
+		}
+
+		#[pallet::call_index(14)]
+		#[pallet::weight(0)]
+		pub fn set_delegation_preferences(
+			origin: OriginFor<T>,
+			parameters: DelegationPreferences,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+			// TODO: Check role operator.
+			OperatorParameters::<T>::insert(account_id, parameters);
+			Ok(())
+		}
+
+		#[pallet::call_index(17)]
+		#[pallet::weight(0)]
+		pub fn block_delegator(origin: OriginFor<T>, delegator_id: T::AccountId) -> DispatchResult {
+			let operator_id = ensure_signed(origin)?;
+			BlockedDelegators::<T>::mutate(&operator_id, |delegators| {
+				delegators.insert(delegator_id.clone());
+			});
+
+			Self::deposit_event(Event::DelegatorBlocked {
+				operator: operator_id,
+				delegator: delegator_id,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::call_index(18)]
+		#[pallet::weight(0)]
+		pub fn unblock_delegator(
+			origin: OriginFor<T>,
+			delegator_id: T::AccountId,
+		) -> DispatchResult {
+			let operator_id = ensure_signed(origin)?;
+			BlockedDelegators::<T>::mutate(&operator_id, |delegators| {
+				delegators.remove(&delegator_id);
+			});
+
+			Self::deposit_event(Event::DelegatorUnblocked {
+				operator: operator_id,
+				delegator: delegator_id,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::call_index(19)]
+		#[pallet::weight(0)]
+		pub fn allow_delegator(origin: OriginFor<T>, delegator_id: T::AccountId) -> DispatchResult {
+			let operator_id = ensure_signed(origin)?;
+			AllowedDelegators::<T>::mutate(&operator_id, |delegators| {
+				delegators.insert(delegator_id.clone());
+			});
+
+			Self::deposit_event(Event::DelegatorAllowed {
+				operator: operator_id,
+				delegator: delegator_id,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::call_index(20)]
+		#[pallet::weight(0)]
+		pub fn disallow_delegator(
+			origin: OriginFor<T>,
+			delegator_id: T::AccountId,
+		) -> DispatchResult {
+			let operator_id = ensure_signed(origin)?;
+			AllowedDelegators::<T>::mutate(&operator_id, |delegators| {
+				delegators.remove(&delegator_id.clone());
+			});
+
+			Self::deposit_event(Event::DelegatorDisallowed {
+				operator: operator_id,
+				delegator: delegator_id,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::call_index(21)]
+		#[pallet::weight(0)]
+		pub fn register_as_operator(origin: OriginFor<T>) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+			T::AccountRoleRegistry::register_as_operator(&account_id)?;
+			Ok(())
+		}
+
+		#[pallet::call_index(22)]
+		#[pallet::weight(0)]
+		pub fn deregister_as_operator(origin: OriginFor<T>) -> DispatchResult {
+			let account_id = T::AccountRoleRegistry::ensure_operator(origin)?;
+
+			ensure!(
+				Self::get_all_validators_by_operator(&account_id).is_empty(),
+				Error::<T>::StillAssociatedWithValidators
+			);
```

Comment:
I know we're not there yet but FWIW the most important check here will be whether there are any active delegations.